saleor/discount/tasks.py

```diff
@@ -232,12 +232,21 @@ def decrease_voucher_codes_usage_task(voucher_code_ids):
     ):
         for voucher_code in voucher_codes:
             if voucher_code.voucher.usage_limit and voucher_code.used > 0:
-                voucher_code.used = F("used") - 1
+                # to ensure backward compatibility we must handle the case
+                # without `codes`
+                if not codes:
+                    used_in_draft_orders = 1
```

Comment:
It should be `id codes is None` if just want to handle this scenario:

> You can have a task in the queue, scheduled by previous implementation, where the new implementation will pick it up.

Reply:
corrected